src/usPrivacy.js

```diff
@@ -0,0 +1,68 @@
+import window from 'global/window';
+
+const findUspApiLocatorWindow = (windowObj) => {
+  let targetWindow = windowObj.parent;
+
+  while (targetWindow !== windowObj.top) {
+    if (targetWindow.frames && targetWindow.frames.__uspapiLocator) {
+      return targetWindow;
+    }
+
+    targetWindow = targetWindow.parent;
+  }
+
+  return windowObj.top;
+};
+
+let uspString = '';
+
+export const getCurrentUspString = () => uspString;
+
+// Call the USP API to get the US Privacy String, either by invoking it directly or via postMessage() if inside an iframe.
+// In the former case the callback is synchronous, if the latter it is asynchronous, so to be safe it should always be
+// assumed to be asynchronous.
+// The window is passable as an argument for ease of testing
+export const obtainUsPrivacyString = (callback, windowObj = window) => {
+  if (windowObj.__uspapi) {
+    windowObj.__uspapi('getUSPData', 1, (uspData, success) => {
+      const privacyString = success ? uspData.uspString : null;
+
+      uspString = privacyString;
+
+      callback(privacyString);
+    });
+  } else {
+    const uniqueId = Math.random().toString(36).substring(2);
+    const message = {
+      __uspapiCall: {
+        command: 'getUSPData',
+        version: 1,
+        callId: uniqueId
+      }
+    };
+
+    const handleMessageEvent = (event) => {
+      if (
+        event &&
+        event.data &&
+        event.data.__uspapiReturn &&
+        event.data.__uspapiReturn.callId === uniqueId
```

Comment:
Random observation: I can't wait until we can start using optional chaining!

Reply:
Iirc last time we looked into that Babel replaced optional chaining in a very verbose way, increasing the dist versions size by a non-trivial amount.

Reply:
The question is whether all the browsers that Video.js 8 supports has native optional chaining, in which case it may not compile it and we could use it.

Reply:
Old Chrome is the problem. Introduced in v80.

Reply:
huh, it's that new?